formats/ome-converter-plugin/src/polus/plugins/formats/ome_converter/image_converter.py

```diff
@@ -0,0 +1,78 @@
+"""Ome Converter."""
+import logging
+import pathlib
+from multiprocessing import cpu_count
+
+from bfio import BioReader, BioWriter
+
+logger = logging.getLogger(__name__)
+
+TILE_SIZE = 2**13
+
+num_threads = max([cpu_count() // 2, 2])
+
+
+def image_converter(
+    inp_image: pathlib.Path, file_extension: str, out_dir: pathlib.Path
+) -> None:
+    """Convert datatypes which are supported by BioFormats to ome.tif or ome.zarr file format.
+
+    Args::
+        inpImage - Path of an input image
+        fileExtension - Type of data conversion
+        outDir - Path to output directory
+    Returns:
+        None
+    """
+    assert file_extension in [
+        ".ome.zarr",
+        ".ome.tif",
+    ], "Invalid fileExtension !! it should be either .ome.tif or .ome.zarr"
+
+    with BioReader(inp_image) as br:
```

Comment:
Not setting the number of workers here could overload the system. If we have `num_cpus` number of threads, and then we allow each thread to have `num_cpus//2` threads for reading and writing, we are effectively trying to use `num_cpus^2 // 2` cpus. We might overload the system and slow things down when we have large numbers of images.

Could you grab something like the MaricRatBrain fovs data and test this out?

Reply:
I have replaced ThreadPoolExecutor with ProcessPoolExecutor and redefined number of workers to be half of the number of logical CPU cores